lib/sensors/gyroscope.cpp

```diff
@@ -0,0 +1,122 @@
+#include "gyroscope.hpp"
+
+namespace hyped::sensors {
+
+Gyroscope::Gyroscope(core::ILogger &logger, io::II2c &i2c, const std::uint8_t channel)
+    : logger_(logger),
+      i2c_(i2c),
+      channel_(channel)
+{
+}
+
+Gyroscope::~Gyroscope()
+{
+}
+
+const std::optional<std::int16_t> Gyroscope::read(core::Axis axis)
+{
+  switch (axis) {
+    case core::Axis::kX: {
+      const auto gyroscope_x_high_byte = i2c_.readByte(kDefaultGyroscopeAddress, kDataXHigh);
+      if (!gyroscope_x_high_byte) {
+        logger_.log(
+          core::LogLevel::kFatal, "Failed to read gyroscope X-axis high at channel %d", channel_);
+        return std::nullopt;
+      }
+      const auto gyroscope_x_low_byte = i2c_.readByte(kDefaultGyroscopeAddress, kDataXLow);
+      if (!gyroscope_x_low_byte) {
+        logger_.log(
+          core::LogLevel::kFatal, "Failed to read gyroscope X-axis low at channel %d", channel_);
+        return std::nullopt;
+      }
+      const auto x_axis = ((gyroscope_x_high_byte.value() << 8) | gyroscope_x_low_byte.value());
```

Comment:
probably just return this instead of sticking it in a variable first, same for y and z

Reply:
nvm i get why this is like this now